src/util_types/mutator_set/ms_membership_proof.rs

```diff
@@ -99,24 +96,55 @@ impl MsMembershipProof {
             "Function must be called with same number of membership proofs and items. Got {} items and {} membership proofs", own_items.len(), membership_proofs.len()
         );
 
-        let new_item_index = mutator_set.aocl.count_leaves();
+        assert!(membership_proofs
```

Comment:
Some of the sanity checks were inserted as a means to track down bugs. In regards to this one, let me see if I can distill it down to a test case.

Reply:
Actually, I'm not sure how to distill this one into a test case because it occurs at the start of the function body. It seems to me like it should be part of a function interface contract, in which case it is okay to assert it where it is now. I don't know what a test case that could cover it would look like.

All of that said, this sanity check was definitely inserted with the objective of tracking down a bug. I inserted essentially the same block of code in more places in the same function, but later removed those.